extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -652,12 +665,20 @@ export class AmpA4A extends AMP.BaseElement {
           });
         }))
         // some() returns an array of which we only need a single value.
-        .then(returnedArray => returnedArray[0]);
+        .then(returnedArray => returnedArray[0], () => {
+          // Rejection occurs if all keys for this provider fail to validate.
+          return Promise.reject(
+              `All keys ${keyInfoSet.serviceName} failed to verify`);
+        });
       });
     }))
     .then(returnedArray => {
       this.protectedEmitLifecycleEvent_('adResponseValidateEnd');
       return returnedArray[0];
+    }, () => {
+      // rejection occurs if all providers fail to verify.
+      this.protectedEmitLifecycleEvent_('adResponseValidateEnd');
+      return null;
```

Comment:
I think this will cause the whole chain to be fulfilled, not rejected.  If I read MDN properly, returning anything other than a reject from a `.then( , onRejected)` block, then the promise is considered resolved.  Is that the semantic you want here?  I'd think you'd want something like `return Promise.reject('No validation service could verify this key')`?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then